src/storage-control/src/client.rs

```diff
@@ -261,6 +261,112 @@ impl Storage {
         self.inner.list_objects().set_parent(parent)
     }
 
+    /// Gets the IAM policy for a specified bucket.
+    ///
+    /// # Parameters
+    /// * `resource` - the bucket name. In `projects/_/buckets/{bucket_id}`
+    ///   format.
+    ///
+    /// # Example
+    /// ```
+    /// # use google_cloud_storage_control::client::Storage;
+    /// async fn example(client: &Storage) -> gax::Result<()> {
+    ///     let policy = client
+    ///         .get_iam_policy("projects/_/buckets/my-bucket")
+    ///         .send()
+    ///         .await?;
+    ///     println!("policy details={policy:?}");
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_iam_policy(
+        &self,
+        resource: impl Into<String>,
+    ) -> super::builder::storage::GetIamPolicy {
+        self.inner.get_iam_policy().set_resource(resource.into())
+    }
+
+    /// Updates the IAM policy for a specified bucket.
+    ///
+    /// This is not an update. The supplied policy will overwrite any existing
+    /// IAM Policy. You should first get the current IAM policy with
+    /// `get_iam_policy()` and then modify that policy before supplying it to
+    /// `set_iam_policy()`.
+    ///
+    /// # Parameters
+    /// * `resource` - the bucket name. In `projects/_/buckets/{bucket_id}`
+    ///   format.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use google_cloud_storage_control::client::Storage;
+    /// async fn example(client: &Storage) -> gax::Result<()> {
+    ///     // Get the current IAM policy for this bucket
+    ///     let policy = client
+    ///         .get_iam_policy("projects/_/buckets/my-bucket")
+    ///         .send()
+    ///         .await?;
+    ///
+    ///     // Add a binding to the policy
+    ///     let mut new_policy = policy.clone();
+    ///     new_policy.bindings.push(
+    ///         iam_v1::model::Binding::new()
+    ///             .set_role("roles/storage.legacyBucketReader")
+    ///             .set_members(["user:me@example.com"]),
+    ///     );
+    ///
+    ///     // Update the IAM policy for this bucket
+    ///     let response = client
+    ///         .set_iam_policy("projects/_/buckets/my-bucket")
+    ///         .set_update_mask(wkt::FieldMask::default().set_paths(["bindings"]))
+    ///         .set_policy(new_policy)
+    ///         .send()
+    ///         .await?;
+    ///     println!("response details={response:?}");
+    ///
+    ///     Ok(())
+    /// }
```

Comment:
This example is too big for the IDE tooltip.  Consider something like:

```rust
    /// async fn example(client: &Storage) -> gax::Result<()> {
    ///     let policy = compute_updated_policy(client, "projects/_/buckets/my-bucket").await?;
    ///     let response = client
    ///         .set_iam_policy("projects/_/buckets/my-bucket")
    ///         .set_policy(policy)
    ///         .set_update_mask(wkt::FieldMask::default().set_paths(["bindings"]))
    ///         .send().await?;
    ///     Ok(())
    /// }
    ///
    /// async fn compute_updated_policy(client: &Storage, bucket: &str) -> gax::Result<Policy> {
    /// # panic!();
    /// }
    ///
```

Reply:
Done.